Exoplanets: Garbage Edition

comma opened the pull request:
Animals will now respawn if killed off too much.
Adds garbage planets, full of rubble and hostile robots. Ripe for xenoarcheology stuff.

PsiOmegaDelta commented on code/game/objects/structures/rubble.dm:
```diff
@@ -0,0 +1,96 @@
+	/obj/item/weapon/archaeological_find/trap)
+
+/obj/structure/rubble/war/New()
+	..()
```
Qué?

comma replied:
oh woops forgot to axe after moving stuff to base one

PsiOmegaDelta commented:
In case it'll be useful, areas have the `turf_initializer` var that you can assign a `/decl/turf_initializer` type to have a given area do things to turfs.
It's how maintenance auto-generate dust, garbage, and other clutter for example.

comma commented:
In this case I'm mostly using noise map's values to pick which tiles get garbage piles so area initializer would be too random.

PsiOmegaDelta commented on code/game/objects/structures/rubble.dm:
```diff
@@ -0,0 +1,93 @@
+	..()
+	if(prob(emptyprob)) 
+		lootleft = 0
+	update_icon()
```
If you could do `update_icon()` in Initialize instead, that'd be appreciated.

comma committed:
Exoplanet additions

Adds garbage exoplanets, desert worlds full of rubble and robots. Rubble holds xenoarch finds.
Biosphere now repopulates itself if more than half of animals are dead.
Moves xenoarch finds from giant switch statement into subtypes.
Adds rubble piles.

PsiOmegaDelta commented on code/modules/xenoarcheaology/finds/finds.dm:
```diff
@@ -7,6 +7,13 @@
 	var/prob_delicate = 90			//probability it requires an active suspension field to not insta-crumble
 	var/dissonance_spread = 1		//proportion of the tile that is affected by this find
 									//used in conjunction with analysis machines to determine correct suspension field type
+/client/verb/butts()
```
I don't think this is a verb we want.

comma replied:
Whoops

PsiOmegaDelta commented:
The bonus client verb aside, not spotting any major issues.
Mostly pondering if the archeological finds shouldn't be subtypes of /obj/random.

comma commented:
They might do with that, but honestly I'd rather not look at that area of code for some time now. Definetly something to consider in future.
